src/components/NotificationList/NotificationList.js

```diff
@@ -0,0 +1,149 @@
+// @flow
+
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+import { Motion, spring } from 'react-motion';
+import Notification from '../Notification';
+import IconBase from 'react-icons-kit';
+
+import { chevronUp } from 'react-icons-kit/feather/chevronUp';
+import { chevronDown } from 'react-icons-kit/feather/chevronDown';
+import { COLORS } from '../../constants';
+
+type ScrollSinkProps = {
+  scrollTop: number,
+  scrollingElementRef: ?HTMLDivElement,
+};
+
+class ScrollSink extends Component<ScrollSinkProps> {
+  componentDidUpdate(prevProps) {
+    const { scrollTop, scrollingElementRef } = this.props;
+    if (scrollingElementRef && prevProps.scrollTop !== scrollTop) {
+      scrollingElementRef.scrollTop = scrollTop;
+    }
+  }
+
+  render() {
+    return null;
+  }
+}
```

Comment:
Interesting approach! I love behavioural components like this; I gave a lightning talk at React Europe last year about how much fun components that return `null` are.

A few notes:

- I'd like to enforce that we only have 1 component per file, unless it's a super trivial stateless-functional-component. Definitely anything with lifecycle methods should get its own file.
- I feel like it breaks the component boundary a bit, to have a component set properties on a different component's HTML element. I feel like if a component is going to manage the scroll position of an element, it should own that element as well.
- I suspect this would have a performance issue on less-powerful machines, as trying to set `scrollTop` is probably pretty expensive, and React Motion will try to run it at 60fps. While we'd lose the nice spring physics animation, I think [`scrollIntoView`](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) would work a lot better.
- If we really wanted to keep the animation, we'd have to use `transform: translate` instead. Essentially how this would work is the parent would be a fixed height, like 60px, and each child would be the same height, 60px. So if you have 5 children, you'd have a 300px list within a 60px container. That 300px list would get its own wrapping <div>. When you click through the notifications, it would offset that wrapper's `translate` by 60px.

I didn't go into a ton of detail for these points since I believe this component will be refactored (either completely replaced by native OS notifications, or not having this behaviour within the notification), but lemme know if you have any questions out of curiosity

Reply:
I'm sad we're not using this system just because this would have been a fun challenge. Native notifications are definitely the way to go though 💯